src/rest-server/src/models/v2/utils/frameworkConverter.js

```diff
@@ -379,7 +466,145 @@ const convertTaskDetail = async (
+      new Date(attemptStatus.completionTime).getTime() || null,
+    containerId: attemptStatus.podUID,
+    containerIp: attemptStatus.podHostIP,
+    containerNodeName: attemptStatus.podNodeName,
+    // Job level info
+    containerPorts,
+    containerGpus,
+    containerLog: `http://${attemptStatus.podHostIP}:${process.env.LOG_MANAGER_PORT}/log-manager/tail/${userName}/${logPathInfix}/${taskRoleName}/${attemptStatus.podUID}/`,
+    containerExitCode: completionStatus ? completionStatus.code : null,
+    containerExitSpec: completionStatus
+      ? generateExitSpec(completionStatus.code)
+      : generateExitSpec(null),
+    containerExitDiagnostics: exitDiagnostics
+      ? exitDiagnostics.diagnosticsSummary
+      : null,
+    ...(launcherConfig.enabledHived && {
+      hived: {
+        affinityGroupName,
+        lazyPreempted: null,
+        lazyPreemptionStatus: null,
+      },
+    }),
+  };
+};
+
+const convertToTaskDetail = async (
+  attemptFramework,
+  taskRoleName,
+  taskStatus,
+  taskHistories,
+) => {
+  const lastTaskAttemptStatus = taskStatus.attemptStatus;
+  const completionStatus = lastTaskAttemptStatus.completionStatus;
+  const userName = attemptFramework.metadata.labels.userName;
+  const jobName = attemptFramework.metadata.annotations.jobName;
+
+  const taskDetail = {
+    // job level information
+    username: userName,
+    jobName: jobName,
+    jobAttemptId: attemptFramework.status.attemptStatus.id,
+    // task role level information
+    taskRoleName: taskRoleName,
+    // task level information
+    taskIndex: taskStatus.index,
+    taskUid: taskStatus.instanceUID,
+    taskState: convertState(
+      taskStatus.state,
+      completionStatus ? completionStatus.code : null,
+    ),
+    retries: taskStatus.retryPolicyStatus.totalRetriedCount,
+    accountableRetries: taskStatus.retryPolicyStatus.accountableRetriedCount,
+    createdTime: new Date(taskStatus.startTime).getTime(),
+    completedTime: new Date(taskStatus.completionTime).getTime(),
+    // task attempt level information
+    attempts: [],
+  };
+
+  const logPathInfix = attemptFramework.metadata.annotations.logPathInfix
+    ? attemptFramework.metadata.annotations.logPathInfix
+    : jobName;
+
+  const ports = attemptFramework.spec.taskRoles.find(
+    (taskRoleSpec) => taskRoleSpec.name === taskRoleName,
+  ).task.pod.metadata.annotations['rest-server/port-scheduling-spec'];
+
+  // fill task attempt level information
+  // last task attempt
+  taskDetail.attempts.push(
+    await convertTaskAttempt(
+      logPathInfix,
+      userName,
+      ports,
+      taskStatus,
+      taskRoleName,
+      lastTaskAttemptStatus,
+    ),
+  );
+
+  // history task attempts
+  for (const taskHistory of taskHistories) {
+    taskDetail.attempts.push(
+      await convertTaskAttempt(
+        logPathInfix,
+        userName,
+        ports,
+        taskStatus,
+        taskRoleName,
+        taskHistory.status.attemptStatus,
+      ),
```

Comment:
await convertTaskAttempt(
        logPathInfix,
        userName,
        ports,
        taskHistory.status,,
        taskRoleName,
        
      ),

Reply:
> ```
>   await convertTaskAttempt(
>     logPathInfix,
>     userName,
>     ports,
>     taskHistory.status,,
>     taskRoleName,
>     
>   ),
> ```

Updated to `await convertTaskAttempt(
    logPathInfix,
    userName,
    ports,
    taskRoleName,
   attemptsStatus,
  ),`